### PR TITLE
Removed tails inherit their color

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

Hey there,

image

Something that was pissing me off was that tails weren't inheriting the color when they were removed from a carbon. I actually did a bit of chicanery in cat surgeon code so I wouldn't be so pissed off about it, but I think now is the time to fix it for real.
## Why It's Good For The Game

image

Much better. I also got rid of one of my pet peeves since I decided to do this in `on_remove()` (nothing should break as a consequence, though let me know if anything looks cringe) and was able to remove the fugly override of parent on `Remove()` which didn't need to be like that at all
## Changelog
:cl:
fix: Carbons with tails (felinids, lizards) who have that tail removed will now have that tail actually look like it came from the person in question, rather than just be a grey thing of sadness.
/:cl:

### CODE REVIEW

**Comment by san7890** [comment]
btw monkeys don't drop tails. there's already an issue report for this but they just don't drop tails. no clue where to even start on that one, and it's out of scope for the intents and purposes of this PR (which is just to fix the color of tails that already drop).

**Comment by Y0SH1M4S73R** [comment]
this could also be applied to other feature organs with colors (snouts and frills)

**Comment by san7890** [comment]
> this could also be applied to other feature organs with colors (snouts and frills)

do those correspond to items? i thought snouts were drawn on the head

File: code/modules/surgery/organs/external/tails.dm [hunk 1]
```diff
@@ -34,13 +34,23 @@
...
 	UnregisterSignal(organ_owner, COMSIG_ORGAN_WAG_TAIL)
 
 	if(type in organ_owner.dna.species.external_organs)
 		organ_owner.add_mood_event("tail_lost", /datum/mood_event/tail_lost)
 		organ_owner.add_mood_event("tail_balance_lost", /datum/mood_event/tail_balance_lost)
 
+	if(bodypart_overlay.color_source == NONE)
+		return
+
+	color = bodypart_overlay.draw_color // so a pink felinid doesn't drop a gray tail
```
Comment (Time-Green) [inline code/modules/surgery/organs/external/tails.dm:52]: Can you move this up to the /external type? Snouts, horns, spines, podhair etc would also benefit from not being gray

**Review by Time-Green — CHANGES_REQUESTED** [review]

**Pull request merged**